runtime/src/main/java/org/corfudb/runtime/view/workflows/WorkflowRequest.java

```diff
@@ -63,8 +64,14 @@ protected ManagementClient getOrchestrator(Layout layout) {
             throw new WorkflowException("getOrchestrator: no available orchestrators " + layout);
         }
 
+        int ind = ThreadLocalRandom.current()
```

Comment:
could you please add comments to this. Especially to the fact that activeLayoutServers list might not be correct as it is not possible to change layout once we have lost quorum. Also another possible approach could be to ping and check which server is up before assigning it.

Reply:
done.